src/node.js

```diff
@@ -22,6 +22,44 @@
 
     EventEmitter.call(process);
 
+    // Default process warning handler..
+    process.on('warning', (warning) => {
+      // By default, print the short warning message if --no-warnings is not
+      // set. If --trace-warnings is set, then the stack trace for the warning
+      // is shown. This takes backwards compatibility with --trace-deprecation
+      // into consideration also.
+      if (process.noProcessWarnings) return;
+      const trace = process.traceProcessWarnings ||
+                    (warning.name === 'DeprecationWarning' &&
+                     process.traceDeprecation);
+      if (trace)
+        console.error(warning.stack);
+      else
+        console.error(`${warning.name}: ${warning.message}`);
+    });
+
+    // process.emitWarning(error)
+    // process.emitWarning(str)
+    // process.emitWarning(options)
+    var Warning;
+    process.emitWarning = function(warning) {
+      if (!Warning)
+        Warning = NativeModule.require('internal/warning');
+      if (typeof warning === 'string') {
+        warning = Warning('Warning', warning, process.emitWarning);
+      } else if (!(warning instanceof Error) && typeof warning === 'object') {
+        if (warning.message) {
```

Comment:
oh.. ha! `null`. sigh, I always forget that one. It's so silly that `typeof null === 'object'`.. sigh